File: uikit/src/main/java/listeners/OnItemClickListener.java
package listeners;

import com.cometchat.pro.models.Group;
import com.cometchat.pro.models.User;

/*

* Purpose - <code>OnItemClickListener&lt;T&gt;</code> is an Generic class used to provide methods
 like <code>OnItemClick(T var,int position)</code> & <code>OnItemLongClick(T var,int position)</code>

* Created on - 20th December 2019

* Modified on  - 16th January 2020

*/

public abstract class OnItemClickListener<T> {

    public abstract void OnItemClick(T var, int position);

    public void OnItemLongClick(T var,int position) {

    }
}

File: uikit-kotlin/src/main/java/listeners/OnItemClickListener.kt
package listeners

/*

* Purpose - <code>OnItemClickListener&lt;T&gt;</code> is an Generic class used to provide methods
 like <code>OnItemClick(T var,int position)</code> & <code>OnItemLongClick(T var,int position)</code>

* Created on - 20th December 2019

* Modified on  - 16th January 2020

*/
abstract class OnItemClickListener<T> {
    abstract fun OnItemClick(t: Any, position: Int)
    fun OnItemLongClick(t: Any, position: Int) {}
}